### ISSUE TITLE
the "spanTo-2" constraint in att.spanning

### ISSUE THREAD

**Issue by bansp** [TEIC/TEI#1624]
I'm wondering about whether the "spanTo-2" constraint (cited below) doesn't attempt to regulate too much, especially now that a separate ticket has been posted on the nature of Schematron constraints in the TEI source.
```xml
<constraintSpec ident="spanTo-2" scheme="isoschematron">
        <desc>The @spanTo attribute must point to an element following the
	current element</desc>
        <constraint>
          <sch:rule context="tei:*[@spanTo]">
            <sch:assert test="id(substring(@spanTo,2)) and following::*[@xml:id=substring(current()/@spanTo,2)]">
The element indicated by @spanTo (<sch:value-of select="@spanTo"/>) must follow the current element <sch:name/>
                  </sch:assert>
          </sch:rule>
        </constraint>
      </constraintSpec>
```

A span has a beginning and an end (yes, I'm brilliant), but I am not sure we want to impose the way a span is defined onto an encoder, the more so that it might happen that in left-to-right contexts, indexing runs right-to-left, while the most natural way to define spans is left-to-right.

Another thing is whether the use of `id()` is fully legitimate here: `@spanTo` is defined at "teidata.pointer", whereas the constraint appears to impose an extra condition on that anyURI (which can be remote in a document that doesn't define IDs in its schema) or which could be an identifier in a local document that is not defined as an ID, but rather expected to be indexed and key()ed for.

In short: shouldn't this constraint be removed?

**Comment by peterstadler** [comment]
I find that constraint very convenient – and I believe it to be in accordance with the note on [att.spanning](http://www.tei-c.org/release/doc/tei-p5-doc/en/html/ref-att.spanning.html) that reads:

> The span is defined as running in document order from the start of the content of the pointing element to the end of the content of the element pointed to by the `spanTo` attribute (if any) …

**Comment by jamescummings** [comment]
I think this should be wont-fix. The schematron implements precisely what we want for the first point of direction of pointing. Whether or not it should really allow anyURI rather than the subset of `#ID-in-Document` is a different matter but I think for the att.spanning elements it shouldn't be going across documents.
